src/base.jl

```diff
@@ -1089,12 +1118,38 @@ function validate_struct(sys::System, value::IS.InfrastructureSystemsType)
     return true
 end
 
+"""
+Check system consistency and validity.
```

Comment:
@claytonpbarrows @jd-lara This function no longer mutates the system. I think that we should remove the `!` in this PR. Do you have any intention of this mutating any attribute in the future? Since this PR exports the functions it will make users wonder what it may change.

Reply:
I think it's ok to remove the `!` here. I think it's better to have separation between the checks and the corrections. Which I suppose answers your second question... I agree that we should change this to `correct`. This would start a newly distinct set of data corrections functionality.

Reply:
@claytonpbarrows `correct_component!` or `sanitize_component`?

Reply:
lets go with `sanitize_component!`

Reply:
do we also need to change `check_angle_limits!` to `sanitize_angle_limits!`?